framework/src/binary_manager/binary_manager_update.c

```diff
@@ -30,7 +30,7 @@
 #include <binary_manager/binary_manager.h>
 #include "binary_manager_internal.h"
 
-binmgr_result_type_e binary_manager_set_bootparam(uint8_t type)
+binmgr_result_type_e binary_manager_set_bootparam(uint8_t type, binary_setbp_result_t *update_result)
```

Comment:
There are 2 options to make api of f/w clearly.
1. you can use binary_setbp_result_t as a type of function instead of binmgr_result_type_e.
2. Refine new struct as a result type which include both of result value.
Checking both of return value & param is not good options as a api of framework. simple is best.

Reply:
I agree with your opinion. But if it returns arrays(binary_setbp_result_t), we need to allocate memory and return it then user should free the result. So I implemented with [out] parameter.
I will consider more later.